LE Audio: CSIP API refactor

Thalley opened the pull request:
Updates the CSIP API to follow the same structure as VCP and MICP. 

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/45625

Thalley committed:
Bluetooth: Audio: Rename CSIS to CSIP

Renamed the coordinated set implementation to use the
profile (CSIP) name instead of the service (CSIS) name, as
the former is more general.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Rename `bt_csip` to `bt_csip_set_member`

Renamed remaining structs and functions that are not
for the set coordinator to use the set member CSIP
role name.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: CSIP set coordinator API not using bt_csip

Modified the CSIP set coordinator slightly to not use the
bt_csip struct at all. For the use a void pointer works
equally well.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Use svc_inst instead of bt_csip for set coordinator

Use the bt_csip_set_coordinator_svc_inst struct in the
set coordinator implementation instead of the more
generic bt_csip.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Use svc_inst instead of bt_csip for set_member

Use the service instance struct instead of the more generic
`bt_csip` for the CSIP set member API.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley commented:
Compliance won't pass due to this bug: https://github.com/zephyrproject-rtos/zephyr/issues/51259

Thalley commented:
Rebased to solve merge conflicts

kruithofa commented on subsys/bluetooth/shell/csip_set_coordinator.c:
```diff
@@ -1,6 +1,6 @@
 /**
  * @file
- * @brief Shell APIs for Bluetooth CSIS client
+ * @brief Shell APIs for Bluetooth CSIP set coordinator
```
you may consider updating the copyright year

kruithofa commented on tests/bluetooth/bsim_bt/bsim_test_audio/test_scripts/csip.sh:
```diff
@@ -4,9 +4,9 @@
-# Basic CSIS test. A set coordinator (CSIS client) connects to multiple set
-# devices (CSIS), lock thems, unlocks them and disconnects.
-SIMULATION_ID="csis_csip"
+# Basic CSIP test. A set coordinator connects to multiple set members
```
you may consider changing the copyright year

kruithofa commented:
> Compliance won't pass due to this bug: #51259

Does it mean this PR will not be merged before that bug is fixed?

Thalley commented:
Rebased to solve merge conflicts

Thalley commented:
> > Compliance won't pass due to this bug: #51259
> 
> Does it mean this PR will not be merged before that bug is fixed?

It means that we will need to either A) override the false-positive or B) rewrite the (otherwise correct code) to pass the check or C) wait for the check to be fixed or D) remove the check from CI

A) will be easiest to do for this PR.

Thalley commented:
Rebased to solve merge conflicts

Thalley commented:
Rebased to solve merge conflicts

Thalley committed:
Bluetooth: Audio: Fix CSIP compliance check with returns after else

The compliance check did not like the otherwise valid
way of returning from read_set_sirk.

The code has been slightly refactored to avoid the check
failing in CI.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley commented:
Got tired of the CI error, so refactored the code.

Thalley commented:
Rebased to solve merge conflicts

MariuszSkamra reviewed (commented):
Did you consider to shorten the `coordinator` to `coord`, and maybe `member` to `mbr`

Thalley commented:
> Did you consider to shorten the `coordinator` to `coord`, and maybe `member` to `mbr`

I've been asking around for abbreviations for those, as I do think they are quite long, but haven't found any good contenders for abbreviations for those 2. 

For example `coord` is used as an abbreviation for "coordinates" in Zephyr, and `mbr` is used for "master boot record". 

Ideally I'd like to use abbreviations that does not clash with other abbreviations in Zephyr.

Thalley commented:
@alwa-nordic @kruithofa Please re-review :)

Thalley commented:
Rebased to solve merge conflicts

Thalley commented:
@alwa-nordic @kruithofa @MariuszSkamra You've all previously approved this. Will you please re-review?
